homeassistant/components/automation/numeric_state.py

```diff
@@ -33,28 +36,43 @@ def async_trigger(hass, config, action):
     entity_id = config.get(CONF_ENTITY_ID)
     below = config.get(CONF_BELOW)
     above = config.get(CONF_ABOVE)
+    time_delta = config.get(CONF_FOR)
     value_template = config.get(CONF_VALUE_TEMPLATE)
+    async_remove_track_same = None
+
     if value_template is not None:
         value_template.hass = hass
 
+    variables = {
+        'trigger': {
+            'platform': 'numeric_state',
+            'below': below,
+            'above': above,
+        }
+    }
+
     @callback
-    def state_automation_listener(entity, from_s, to_s):
-        """Listen for state changes and calls action."""
+    def check_numeric_state(entity, from_s, to_s):
+        """Return True if they should trigger."""
         if to_s is None:
-            return
+            return False
 
-        variables = {
-            'trigger': {
-                'platform': 'numeric_state',
-                'entity_id': entity,
-                'below': below,
-                'above': above,
-            }
-        }
+        var = deepcopy(variables)
```

Comment:
I think that just keeping the variable declaration inline is just fine. Yeah it is in two places but I feel like deepcopy is a bit of overkill.

Reply:
(bit nitpicky, I know, you can ignore :) )

Reply:
No, you are the founder ;)